kpatch-build/create-diff-object.c

```diff
@@ -2825,6 +2825,21 @@ static void kpatch_mark_ignored_sections(struct kpatch_elf *kelf)
 		    !strncmp(sec->name, ".llvm_addrsig", 13) ||
 		    !strncmp(sec->name, ".llvm.", 6))
 			sec->ignore = 1;
+
+		/*
+		 * With CONFIG_LTO_CLANG, we see some weird new functions,
+		 * such as:
+		 * __initstub__kmod_syscall__728_5326_bpf_syscall_sysctl_init7
+		 * while the original function is very similar, like:
+		 * __initstub__kmod_syscall__728_5324_bpf_syscall_sysctl_init7
+		 *
+		 * We don't have very good solution for it yet. To workaround
+		 * the issue, it seems safe to ignore .init.text changes in
+		 * vmlinux.o. This will skip such new functions.
+		 */
+		if (!strncmp(childobj, "vmlinux.o", 9) &&
+		    !strncmp(sec->name, ".init.text", 10))
+			sec->ignore = 1;
```

Comment:
Also this (or whatever the change eventually is) should probably be in its own commit.